src/vaults/SlippageVault.sol

```diff
@@ -0,0 +1,231 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.16;
+
+import {MathUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/math/MathUpgradeable.sol";
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
+
+import {Vault} from "src/vaults/Vault.sol";
+import {VaultV2} from "src/vaults/VaultV2.sol";
+import {NftGate} from "src/vaults/NftGate.sol";
+import {HarvestStorage} from "src/vaults/HarvestStorage.sol";
+import {BaseStrategy as Strategy} from "src/strategies/BaseStrategy.sol";
+import {uncheckedInc} from "src/libs/Unchecked.sol";
+import {VaultErrors} from "src/libs/VaultErrors.sol";
+
+contract SlippageVault is VaultV2 {
+    using SafeTransferLib for ERC20;
+    using MathUpgradeable for uint256;
+
+    /**
+     * @dev See {IERC4262-deposit}.
+     */
+    function deposit(uint256 assets, address receiver) public virtual override whenNotPaused returns (uint256) {
+        _harvestAll();
+        uint256 shares = previewDeposit(assets);
+        uint256 oldSupply = totalSupply();
+        _deposit(_msgSender(), receiver, assets, shares);
+        return totalSupply() - oldSupply;
+    }
+
+    /**
+     * @dev See {IERC4262-mint}.
+     */
+    function mint(uint256 shares, address receiver) public virtual override whenNotPaused returns (uint256) {
+        _harvestAll();
+        uint256 assets = previewMint(shares);
+
+        uint256 oldTVL = vaultTVL();
+        _deposit(_msgSender(), receiver, assets, shares);
+
+        return vaultTVL() - oldTVL;
+    }
+
+    function _deposit(address caller, address receiver, uint256 assets, uint256 shares) internal virtual override {
```

Comment:
We  can't use `_depositIntoStrategies`. We need to increment totalStrategyHoldings by the exact amount that actually got into the strategies.